sinks/channel/channel_sink.go

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package channel
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/eventlogger"
+)
+
+// ChannelSink is a sink node which sends
+// the event to a channel
+type ChannelSink struct {
+	eventChan chan<- *eventlogger.Event
+
+	// The time to wait for a write before returning an error
+	timeoutDuration time.Duration
+}
+
+var _ eventlogger.Node = &ChannelSink{}
+
+// newChannelSink creates a ChannelSink
+func NewChannelSink(c chan<- *eventlogger.Event, t time.Duration) (*ChannelSink, error) {
```

Comment:
mind adding a comment to the godoc about how the time.Duration is used?

Reply:
Sure!